train-system/src/main/java/hu/bme/mit/train/system/TrainSystem.java

```diff
@@ -7,11 +7,19 @@
 import hu.bme.mit.train.sensor.TrainSensorImpl;
 import hu.bme.mit.train.user.TrainUserImpl;
 
+import java.util.Timer;
+
 public class TrainSystem {
 
 	private TrainController controller = new TrainControllerImpl();
 	private TrainUser user = new TrainUserImpl(controller);
 	private TrainSensor sensor = new TrainSensorImpl(controller, user);
+	
+	private Timer timer = new Timer();
+	
+	public TrainSystem() {
+		timer.scheduleAtFixedRate(controller.followSpeed, 100, 100);
```

Comment:
This line doesn't compile. It thinks followSpeed is a variable.
/TrainSystem.java:21: error: cannot find symbol
timer.scheduleAtFixedRate(controller.followSpeed, 100, 100);
	                                                   ^
symbol:   variable followSpeed
location: variable controller of type TrainController

Reply:
Maybe try it as a lambda?